datastore/client.go

```diff
@@ -44,13 +45,18 @@ type datastoreClient struct {
 
 func newDatastoreClient(conn grpc.ClientConnInterface, projectID, databaseID string) pb.DatastoreClient {
 	resourcePrefixValue := "projects/" + projectID
-	if databaseID != "" {
+	reqParamsHeaderValue := "project_id=" + url.QueryEscape(projectID)
+
+	if databaseID != DefaultDatabaseID {
 		resourcePrefixValue += "/databases/" + databaseID
+		reqParamsHeaderValue += "&database_id=" + databaseID
```

Comment:
should be wrapped in `url.QueryEscape`?

Reply:
Done

Reply:
sorry not sure if I missed something, shouldn't `reqParamsHeaderValue` be escaped?

Reply:
Done